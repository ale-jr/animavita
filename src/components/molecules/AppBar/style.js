import styled from "styled-components";

export const AppBarBox = styled.header`
  width: 100%;

  padding: ${({ theme }) => theme.spacing.xs};
  background-color: ${({ theme }) => theme.colors.primary.medium};
  color: ${({ theme }) => theme.colors.primary.contrast};
`;

export const AppBarTitle = styled.p`
  font-size: ${({ theme }) => theme.typography.sizes.md};
  font-weight: ${({ theme }) => theme.typography.weights.bold};
`;
