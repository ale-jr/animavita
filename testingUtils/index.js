import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../src/themes/default";

function ThemeProviderWrapper({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  );
}

export const renderWithTheme = (ui, options) =>
  render(ui, {
    wrapper: ThemeProviderWrapper,
    ...options
  });
