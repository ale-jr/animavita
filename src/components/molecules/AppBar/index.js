import PropTypes from "prop-types";
import { AppBarBox, AppBarTitle } from "./style";

function AppBar({ title }) {
  return (
    <AppBarBox>
      <AppBarTitle>
        {title}
      </AppBarTitle>
    </AppBarBox>
  );
}

AppBar.propTypes = {
  title: PropTypes.string.isRequired
};

export default AppBar;
