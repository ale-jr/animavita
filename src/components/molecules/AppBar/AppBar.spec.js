import { renderWithTheme } from "../../../../testingUtils";
import AppBar from ".";

describe("AppBar", () => {
  it("renders correclty", () => {
    const { container } = renderWithTheme(<AppBar title="AppBarTitle" />);
    expect(container).toMatchSnapshot();
  });
});
