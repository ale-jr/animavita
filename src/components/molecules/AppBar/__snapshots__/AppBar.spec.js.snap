// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar renders correclty 1`] = `
<div>
  <header
    class="sc-bdvvtL iLoynf"
  >
    <p
      class="sc-gsDKAQ bvmJiY"
    >
      AppBarTitle
    </p>
  </header>
</div>
`;
